src/common/xo/index.js

```diff
@@ -1875,6 +1877,26 @@ export const deleteSshKey = key =>
     })
   }, noop)
 
+export const deleteSshKeys = keys =>
+  confirm({
+    title: _('deleteSshKeysConfirm', { nKeys: keys.length }),
+    body: _('deleteSshKeysConfirmMessage', {
+      nKeys: keys.length,
+    }),
+  }).then(() => {
+    const { preferences } = xo.user
+    return Promise.all(
+      map(keys, key =>
+        _setUserPreferences({
+          sshKeys: filter(
+            preferences && preferences.sshKeys,
+            k => !isEqual(k, key)
```

Comment:
Don't set the user preferences for each key. Instead, build the new sshKeys array and set the preferences once.

Reply:
Have you tested this? I don't think this will work since there is an `id` property in `key` but not in `k`.
You should compare the key strings instead as it is assumed that they should be different. You'll also have to change the `deleteSshKey` method for the same reason.